### ISSUE TITLE
pnpm update does not update package.json when a peer dependency is missing

### ISSUE THREAD

**Issue by val1984** [pnpm/pnpm#4763]
### Code to reproduce the issue:

A package.json with a dependency that has peerDependencies which are not satisfied (here jquery & popper.js because we would be using reactstrap instead):
```
"dependencies": {
  "bootstrap": "^4.6.0"
}
```
Then update (`pnpm up`).

### Expected behavior:
pnpm updates package.json with bootstrap@^4.6.1.

### Actual behavior:
pnpm does not update package.json which remains with bootstrap@^4.6.0. This makes CI install fail with:
```
Lockfile is up-to-date, resolution step is skipped
 ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up-to-date with package.json

Note that in CI environments this setting is true by default. If you still need to run install in such cases, use "pnpm install --no-frozen-lockfile"
```

### Workaround:
`pnpm up --config.strict-peer-dependencies=false`

### Additional information:
 - `node -v` prints:
 `v16.15.0`
 - Windows, macOS, or Linux?:
 Windows

**Comment by ghostdevv** [comment]
+1

**Comment by mcmxcdev** [comment]
I think this works as expected because `strict-peer-dependencies` is enabled by default since pnpm v7 and therefore `pnpm up`  exits with ` ERR_PNPM_PEER_DEP_ISSUES  Unmet peer dependencies`.

There is even a `hint: If you don't want pnpm to fail on peer dependency issues, set the "strict-peer-dependencies" setting to "false".` to notify you.

So you can opt-out of that intentional behavior with the "workaround" you mentioned above.

**Comment by val1984** [comment]
I think that there should be some kind of explicit warning saying that `package.json` file has not being updated because of this new default setting, considering there are at least 3 issues about ppm update no longer updating package.json.

**Comment by ghostdevv** [comment]
also it's updating the lock, if it fails it shouldn't fail on one but not the other surely?

**Comment by mcmxcdev** [comment]
I agree with both of you.

**Comment by val1984** [comment]
> also it's updating the lock, if it fails it shouldn't fail on one but not the other surely?

Sorry to ping you @zkochan but what's your take on this issue and particularly the quoted message?

**Comment by zkochan** [comment]
I agree. `package.json` should probably be updated even if it fails.

**Comment by ptandler** [comment]
In my case the peer dependency was prettier (we use v2, the package uses v1).

If package.json and lock is not updated, there should be at least a messaging explaining what's going on and how I can work around.

**Comment by val1984** [comment]
> I agree. `package.json` should probably be updated even if it fails.

@zkochan Any progress on this matter? Thank you in advance!

**Comment by val1984** [comment]
I haven't noticed this issue anymore with pnpm 8 and above so this issue might as well be closed.

**Comment by xxKeefer** [comment]
I am still experiencing this issue on v8.10.5
